src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextJsonReader.cs

```diff
@@ -656,7 +653,7 @@ private LibraryStub ReadOneLibrary(ref Utf8JsonReader reader)
             return new LibraryStub()
             {
                 Hash = hash,
-                Type = Pool(type),
+                Type = Pool(type)!,
```

Comment:
Why is this valid?

Reply:
Not sure, but it looks like it is expected not to be `null`. Added `Debug.Assert` to verify.